src/views.py

```diff
@@ -16,6 +17,66 @@
 }
 
 
+def sort_by_building(building: str) -> Tuple[int, str]:
+    match = re.match(r"(\d+)(\D*)", building)
+    if match:
+        return (int(match.group(1)), match.group(2))
+    return (0, "")
+
+
+def get_classrooms(request: HttpRequest) -> HttpResponse:
+    form = ClassroomForm(request.POST)
+    if not form.is_valid():
+        return HttpResponseBadRequest()
+    data = form.cleaned_data
+    # Get all bldgs + rooms that have conflicting blocks
+    query = (
+        ScheduleBlock.objects.filter(
+            start_time__lt=data["end_time"],
+            end_time__gt=data["start_time"],
+            day_of_the_week=DAY_OF_THE_WEEK_MAPPING[int(data["day"])],
+        )
+        .values("building", "room")
+        .distinct()
+    )
+    used_rooms = list(query)
+    # Find all bldg+rooms and filter by used rooms
+    all_classrooms = list(
+        ScheduleBlock.objects.all().values("building", "room").distinct()
+    )
+
+    classrooms = list(
+        filter(
+            lambda classroom: classroom not in used_rooms,
+            all_classrooms,
+        )
+    )
+
+    # Create dictionary mapping each building to list of rooms
+    class_map: dict[str, List[str]] = dict()
+    for classroom in classrooms:
+        building = classroom["building"]
+        room = classroom["room"]
+
+        if building not in class_map:
+            class_map[building] = []
+        class_map[building].append(room)
+    # Sort classes
```

Comment:
is this needed anymore if its sorted in return statement?

Reply:
this sorts the classrooms, the return statement sorts the buildings.